mergify_engine/utils.py

```diff
@@ -314,3 +315,22 @@ def split_list(
     while remaining:
         yield remaining[:size]
         remaining = remaining[size:]
+
+
+def async_cache(
+    user_function: typing.Callable[..., typing.Awaitable[_T]]
+) -> typing.Callable[..., typing.Awaitable[_T]]:
+    cache: typing.Dict[typing.Hashable, _T] = {}
```

Comment:
rather use the cache object from functools?
it has more stuff like stats etc.

Reply:
I tried to keep it simple, for stats we need to catch miss/hit ourselves and fill the `functools._CacheInfo()` ourselves.